compiler/quilt/test/test_command.py

```diff
@@ -574,6 +592,57 @@ def test_export(self):
                 # data matches
                 assert shash(export_path.read_bytes()) == shash(install_path.read_bytes())
 
+            # Test force=True
+            # We just tested that calling this raises CommandException with 'file already exists',
+            # so it's a good spot to check the force option.
+            command.export(pkg_name, str(test_dir), force=True)
+
+            exported_paths = [path for path in test_dir.glob('**/*') if path.is_file()]
+
+            assert len(exported_paths) == len(test_data)
+
+            for path, data in test_data:
+                export_path = test_dir / path
+                install_path = install_dir / path
+
+                # filename matches
+                assert export_path in exported_paths
+                # data matches
+                assert shash(export_path.read_bytes()) == shash(install_path.read_bytes())
+
+            # Test raise when exporting to overwrite existing files
+            files = set(f for f in test_dir.glob('*') if f.is_file())
+            # sorted and reversed means files before their containing dirs
+            for path in sorted(test_dir.glob('**/*'), reverse=True):
+                # keep files from root of test_dir
+                if path in files:
+                    continue
+                # remove everything else
+                path.rmdir() if path.is_dir() else path.unlink()
+            # now there are only files in the export root to conflict with.
+            with pytest.raises(command.CommandException, match='file already exists'):
+                command.export(pkg_name, str(test_dir))
+
+            # Test raise when exporting to existing dir structure
```

Comment:
Ditto re Force

Reply:
The existing test would raise the CommandException on either of these cases, because it writes over a full prior install with both subdirs and files -- should I just catch that (potential) error and reraise as a Failed() exception via pytest.fail()?

Reply:
sounds like export() needs to be changed?  export(force=True) should not throw an exception imho... it should just overwrite.

sorry if i'm missing somethg

Reply:
Sorry, to be clear:  The checks you requested already implicitly exist due the the `force=True` test -- because, as you say, quilt export shouldn't raise an exception, and doesn't, currently. 

Both of the checks you noted were to ensure that the non-force version *does* raise those exceptions, but the force version of the test doesn't catch any errors, it just checks to make sure the export worked -- so if any errors are raised, the test fails, as it should.